### PR TITLE
Initialize package and docs for log-collection library transfer

### PR DESCRIPTION
**Pull request by djaglowski**
This PR is a first incremental step towards moving the log-collection library into this repo, as defined by #8850.

An all-in-one PR (#9227) was made, but proved to require many untracked changes. I am now taking an incremental approach as described in [this comment on #9227](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9227#issuecomment-1099681639).

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

File: pkg/stanza/docs/operators/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+## What is an operator?
+An operator is the most basic unit of log processing. Each operator fulfills a single responsibility, such as reading lines from a file, or parsing JSON from a field. Operators are then chained together in a pipeline to achieve a desired result.
+
+For instance, a user may read lines from a file using the `file_input` operator. From there, the results of this operation may be sent to a `regex_parser` operator that creates fields based on a regex pattern. And then finally, these results may be sent to a `elastic_output` operator that writes each line to Elasticsearch.
```
Comment (pmm-sumo) [inline pkg/stanza/docs/operators/README.md:4]:
```suggestion
For instance, a user may read lines from a file using the `file_input` operator. From there, the results of this operation may be sent to a `regex_parser` operator that creates fields based on a regex pattern. And then finally, these results may be sent to a `stdout` operator that writes each line to stdout.
```

This is probably minor, but elastic is no longer included among the outputs. Also, I'm wondering if we should describe that the library is used in the receiver which means it does not need to use outputs normally

File: pkg/stanza/docs/operators/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+## What is an operator?
```
Comment (pmm-sumo) [inline pkg/stanza/docs/operators/README.md:1]: Perhaps we could add a status describing that the library move is in progress?

**Review by pmm-sumo — APPROVED** [review]
Some minor comments, otherwise LGTM

File: pkg/stanza/docs/operators/README.md (hunk 1 above)
Comment (djaglowski) [reply pkg/stanza/docs/operators/README.md]: Good feedback, this doc is quite out of date. I took a pass through the document and updated the contents to reflect the current state much more accurately.

**Comment by djaglowski** [comment]
> @djaglowski one approach we did when moved lots of core components to contrib, was to move everything using git subtree to save history, and explicitly remove that module from the "Makefile" build/test/checks until we were done with it. The beauty was that we saved the entire history of commits.

This sounds like a better approach. I'll look into it.

**Comment by djaglowski** [comment]
I'm converting this back to a draft until Bogdan's suggested approach is looked into.

**Comment by djaglowski** [comment]
Closing in favor of #9376

**Pull request closed without merging**